token/program/src/state.rs

```diff
@@ -418,11 +430,17 @@ mod tests {
         let result = Account::unpack_account_owner(&src);
         assert_eq!(result, Option::None);
 
-        // The right account data size, unpack will return some key
-        let src: [u8; Account::LEN] = [0; Account::LEN];
+        // The right account data size and intialized, unpack will return some key
+        let mut src: [u8; Account::LEN] = [0; Account::LEN];
+        src[ACCOUNT_INITIALIZED_INDEX] = 1;
         let result = Account::unpack_account_owner(&src);
         assert!(result.is_some());
```

Comment:
nit: is it worth having a test case for a frozen account?

Reply:
Ah yes, that's a great idea.  One of my previous suggestions on this PR was actually wrong because I forgot about the frozen state.